### PR TITLE
CRM-20441: Fatal error on contact summary for ACL'd user (from activity tab count)

### PR DESCRIPTION
**Pull request by monishdeb**
* [CRM-20441: Fatal error on contact summary for ACL'd user \(from activity tab count\)](https://issues.civicrm.org/jira/browse/CRM-20441)

### CODE REVIEW

File: CRM/Activity/BAO/Activity.php [hunk 1]
```diff
@@ -788,7 +788,14 @@ public static function getActivities($params, $getCount = FALSE) {
...
     //    current logged in user has permission to view Case activities we are performing filtering out those activities from list (see below).
     //    This logic need to be incorporated in Activity.get definition
-    $result = civicrm_api3('Activity', 'Get', $activityParams);
+    try {
+      $result = civicrm_api3('Activity', 'Get', $activityParams);
+    }
+    catch (CiviCRM_API3_Exception $e) {
+      // CRM-20441: if the loggedin user has not enough permission to view activities
+      //  then return empty result instead of throwring error
+      return $getCount ? count($activities) : $activities;
```
Comment (seamuslee001) [inline CRM/Activity/BAO/Activity.php]: @monishdeb why not return it earlier as in my suggestion?

**Comment by davejenx** [comment]
After reverting my cherry-pick of #10237, I cherry-picked 6b010c01f08f41447bcac38df76364d0596059cf and it likewise fixes the fatal error. :-) Contact summary now displays, with Activities tab showing count = 0.

As a further test, I tried adding an activity to one of the allowed contacts that didn't have any activities. I set source = one of the allowed contacts, target = another of the allowed contacts, no assignee. So that should be visible to the restricted user - and it is: count = 1 on Activities tab, can view activity.

So this fix looks equally as good as #10237 from my testing.

File: CRM/Activity/BAO/Activity.php (hunk 1 above)
Comment (monishdeb) [inline CRM/Activity/BAO/Activity.php]: Ahh right, my 4.7.19-rc branch was outdated and it didn't had that ```try-catch``` code block.  Yup you are right.

**Comment by monishdeb** [comment]
Closing in favor of #10237

**Pull request closed without merging**